js/utils/network-permission.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * Network Permission Manager
+ * Modern MetaMask chain permissions handler using centralized config
+ * 
+ * This utility handles network permissions instead of forcing network switches,
+ * allowing users to stay on any network while the dApp uses the required network.
+ */
+
+class NetworkPermission {
+    /**
+     * Get the expected chain ID from centralized config
+     * @returns {number} Chain ID (e.g., 80002 for Polygon Amoy)
+     */
+    static get AMOY_CHAIN_ID() {
+        return window.CONFIG?.NETWORK?.CHAIN_ID || 80002;
+    }
+
+    /**
+     * Get the expected chain ID in hex format
+     * @returns {string} Chain ID in hex (e.g., '0x13882')
+     */
+    static get AMOY_CHAIN_ID_HEX() {
+        return '0x' + NetworkPermission.AMOY_CHAIN_ID.toString(16);
+    }
+
+    /**
+     * Get network configuration from centralized config
+     * @returns {object} Network configuration
+     */
+    static getNetworkConfig() {
+        return {
+            chainId: NetworkPermission.AMOY_CHAIN_ID_HEX,
+            chainName: window.CONFIG?.NETWORK?.NAME || 'Polygon Amoy Testnet',
+            rpcUrls: [
+                window.CONFIG?.NETWORK?.RPC_URL || 'https://rpc-amoy.polygon.technology',
+                ...(window.CONFIG?.NETWORK?.FALLBACK_RPCS || [])
+            ],
+            nativeCurrency: window.CONFIG?.NETWORK?.NATIVE_CURRENCY || {
+                name: 'MATIC',
+                symbol: 'MATIC',
+                decimals: 18
+            },
+            blockExplorerUrls: [
+                window.CONFIG?.NETWORK?.BLOCK_EXPLORER || 'https://amoy.polygonscan.com'
+            ]
+        };
+    }
+
+    /**
+     * Check if we have permission to use the Amoy network
+     * @returns {Promise<boolean>} True if wallet is connected and on Amoy network
+     */
+    static async hasAmoyPermission() {
```

Comment:
this and the other amoy specific functions could be more generic. The current code would require a refactor when we switch to mainnet